hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/JetMetricsService.java

```diff
@@ -111,21 +116,41 @@ public void init(NodeEngine nodeEngine, Properties properties) {
         }, 1, config.getCollectionIntervalSeconds(), TimeUnit.SECONDS);
     }
 
+    public LiveOperationRegistry getLiveOperationRegistry() {
+        return liveOperationRegistry;
+    }
+
+    @Override
+    public void populate(LiveOperations liveOperations) {
+        liveOperationRegistry.populate(liveOperations);
+    }
+
     /**
      * Read metrics from the journal from the given sequence
      */
-    public ConcurrentArrayRingbuffer.RingbufferSlice<Map.Entry<Long, byte[]>> readMetrics(long startSequence) {
+    public CompletableFuture<RingbufferSlice<Map.Entry<Long, byte[]>>> readMetrics(long startSequence) {
         if (!config.isEnabled()) {
             throw new IllegalArgumentException("Metrics collection is not enabled");
         }
-        return metricsJournal.copyFrom(startSequence);
+        CompletableFuture<RingbufferSlice<Map.Entry<Long, byte[]>>> future = new CompletableFuture<>();
+        future.whenComplete((s, e) -> futureMap.remove(future));
```

Comment:
it's better to use `ExceptionUtil.withTryCatch` for all whenComplete calls, otherwise any exception gets swallowed.